File: app/src/main/java/net/veldor/flibustaloader/adapters/BookmarksAdapter.java
package net.veldor.flibustaloader.adapters;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

import net.veldor.flibustaloader.App;
import net.veldor.flibustaloader.BR;
import net.veldor.flibustaloader.R;
import net.veldor.flibustaloader.database.dao.BookmarksDao;
import net.veldor.flibustaloader.database.entity.Bookmark;
import net.veldor.flibustaloader.databinding.BookmarkItemBinding;
import net.veldor.flibustaloader.ui.OPDSActivity;

import java.util.ArrayList;
import java.util.List;

import static android.content.Intent.FLAG_ACTIVITY_CLEAR_TOP;
import static android.content.Intent.FLAG_ACTIVITY_NEW_TASK;

public class BookmarksAdapter extends RecyclerView.Adapter<BookmarksAdapter.ViewHolder> {
    private List<Bookmark> mBookmarks;
    private LayoutInflater mLayoutInflater;
    private final BookmarksDao mDao;

    public BookmarksAdapter(List<Bookmark> bookmarksList) {
        mBookmarks = bookmarksList;
        mDao = App.getInstance().mDatabase.bookmarksDao();
    }

    @NonNull
    @Override
    public BookmarksAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        if (mLayoutInflater == null) {
            mLayoutInflater = LayoutInflater.from(viewGroup.getContext());
        }
        BookmarkItemBinding binding = DataBindingUtil.inflate(mLayoutInflater, R.layout.bookmark_item, viewGroup, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull BookmarksAdapter.ViewHolder viewHolder, int i) {
        viewHolder.bind(mBookmarks.get(i));
    }

    @Override
    public int getItemCount() {
        return mBookmarks.size();
    }


    public void setContent(ArrayList<Bookmark> arrayList) {
        mBookmarks = arrayList;
        notifyDataSetChanged();
    }


    class ViewHolder extends RecyclerView.ViewHolder {

        private final ViewDataBinding mBinding;

        ViewHolder(ViewDataBinding binding) {
            super(binding.getRoot());
            mBinding = binding;
        }

        void bind(final Bookmark bookmark) {
            mBinding.setVariable(BR.bookmark, bookmark);
            mBinding.executePendingBindings();
            // добавлю действие при клике на кнопку скачивания
            View container = mBinding.getRoot();
            View itemName = container.findViewById(R.id.bookmark_name);
            itemName.setOnClickListener(v -> {
                Intent intent = new Intent(App.getInstance(), OPDSActivity.class);
                intent.putExtra(OPDSActivity.TARGET_LINK, bookmark.link);
                intent.setFlags(FLAG_ACTIVITY_NEW_TASK|FLAG_ACTIVITY_CLEAR_TOP);
                App.getInstance().startActivity(intent);
            });

            View deleteItemBtn = container.findViewById(R.id.deleteItemBtn);
            deleteItemBtn.setOnClickListener(v -> {
                mDao.delete(bookmark);
                mBookmarks.remove(bookmark);
                notifyDataSetChanged();
                Toast.makeText(App.getInstance(), "Закладка удалена", Toast.LENGTH_SHORT).show();
            });
        }
    }
}

File: app/src/main/java/net/veldor/flibustaloader/adapters/BookmarksAdapter.kt
package net.veldor.flibustaloader.adapters

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import net.veldor.flibustaloader.App
import net.veldor.flibustaloader.BR
import net.veldor.flibustaloader.R
import net.veldor.flibustaloader.database.dao.BookmarksDao
import net.veldor.flibustaloader.database.entity.Bookmark
import net.veldor.flibustaloader.databinding.BookmarkItemBinding
import net.veldor.flibustaloader.ui.BrowserActivity
import net.veldor.flibustaloader.ui.fragments.OpdsFragment.Companion.TARGET_LINK

class BookmarksAdapter(private var mBookmarks: MutableList<Bookmark>) :
    RecyclerView.Adapter<BookmarksAdapter.ViewHolder>() {
    private var mLayoutInflater: LayoutInflater = LayoutInflater.from(App.instance.applicationContext)
    private val mDao: BookmarksDao = App.instance.mDatabase.bookmarksDao()
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val binding = BookmarkItemBinding.inflate(mLayoutInflater, viewGroup, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.bind(mBookmarks[i])
    }

    override fun getItemCount(): Int {
        return mBookmarks.size
    }

    inner class ViewHolder(private val mBinding: BookmarkItemBinding) : RecyclerView.ViewHolder(
        mBinding.root
    ) {
        fun bind(bookmark: Bookmark) {
            mBinding.setVariable(BR.bookmark, bookmark)
            mBinding.executePendingBindings()
            // добавлю действие при клике на кнопку скачивания
            val container = mBinding.root
            val itemName = container.findViewById<View>(R.id.bookmark_name)
            itemName.setOnClickListener {
                val intent = Intent(App.instance, BrowserActivity::class.java)
                intent.putExtra(TARGET_LINK, bookmark.link)
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
                App.instance.startActivity(intent)
            }
            val deleteItemBtn = container.findViewById<View>(R.id.deleteItemBtn)
            deleteItemBtn.setOnClickListener {
                mDao.delete(bookmark)
                notifyItemRemoved(mBookmarks.indexOf(bookmark))
                mBookmarks.remove(bookmark)
                Toast.makeText(App.instance, "Закладка удалена", Toast.LENGTH_SHORT).show()
            }
        }
    }

}